src/prog_models/sim_result.py

```diff
@@ -72,7 +98,21 @@ def index(self, other: dict, *args, **kwargs) -> int:
         Returns:
             int: Index of first sample where other occurs
         """
+        warn('index will be deprecated after version 1.5 of ProgPy. The function will be renamed, indexofdata, and users may begin using it under this name now.',
+            DeprecationWarning, stacklevel=2)
+
         return self.data.index(other, *args, **kwargs)
+    def indexofdata(self, other: dict, *args, **kwargs) -> int:
+        """
+        Get the index of the first sample where other occurs
+
+        Args:
+            other (dict)
+
+        Returns:
+            int: Index of first sample where other occurs
+        """
+        return self.index(other, *args, **kwargs)
```

Comment:
Same as pop_by_index. This will still raise the deprecation warning even when they use index_of_data. Move the logic from index into here and have index call this function